cmd/butil/main.go

```diff
@@ -291,3 +301,122 @@ func dodelete(r *items.Store, id string, delblobs []string) {
 		fmt.Println(err.Error())
 	}
 }
+
+// doidentifymissingblob will scan an entire store and list the item ids that
+// have "missing blobs". Passing those ids to fix-missing-blob will repair them.
+//
+// See DLTP-1676.
+func doidentifymissingblobs(r *items.Store) {
+	ids := r.List()
+
+	for id := range ids {
+		item, err := r.Item(id)
+		if err != nil {
+			fmt.Println(err)
+			return
+		}
+		n := len(item.Blobs)
+		if n == 0 {
+			// no blobs?
+			continue
+		}
+		if items.BlobID(n) == item.Blobs[n-1].ID {
+			// nothing missing
+			continue
+		}
+		// missing blobs are present
+		fmt.Println(id)
+	}
+}
+
+type missingPair struct {
+	ID     items.BlobID
+	Bundle int
+}
+
+var (
```

Comment:
After puzzling over this for a minute, I did an MD5 on a zero-length file, Duh. Perhaps a comment for the dense among us, a comment saying the md5 null  hash  is d41d8cd98f00b204e9800998ecf8427e, etc?

Reply:
no prob. I'll add something.